### DISCUSSION TITLE
When compiling a sketch AllProtocols.ino, I receive a message: AllProtocols.ino.elf section `.text1' will not fit in region `iram1_0_seg'

### DISCUSSION THREAD

**Discussion by saharesp** [ukw100/IRMP#93] [Q&A]
When compiling a sketch AllProtocols.ino on ESP8266, I receive a message: AllProtocols.ino.elf section `.text1' will not fit in region `iram1_0_seg'. Compilation occurs only when commenting on the irmp.init() line; in the section: void setup. As I understand it, the IRAM memory is overflowing with global variables. Help me deal with this problem.

**Comment by ArminJo** [comment]
Sorry, this is a known problem, but not well documented until now.
For **ESP8266/ESP32**, [this library](https://github.com/crankyoldgit/IRremoteESP8266) supports an [impressive set of protocols and a lot of air conditioners](https://github.com/crankyoldgit/IRremoteESP8266/blob/master/SupportedProtocols.md)

**Comment by saharesp** [comment]
Thanks for the reply. I am using the library from your link and it works. I wanted to try the IRMP library, as it claims a lot of IR protocols.

    On Tuesday, April 23, 2024 at 04:04:35 PM GMT+9, Armin ***@***.***> wrote:  
 
 


Closed #93 as resolved.
